tests/test_dataclasses.py

```diff
@@ -380,3 +380,28 @@ class Outer:
             }
         },
     }
+
+
+def test_initvar():
+    from math import pi
+
+    InitVar = dataclasses.InitVar
+
+    @pydantic.dataclasses.dataclass
+    class RadiansWithInitVar:
+        rad: float = dataclasses.field(init=False)
+        deg: InitVar
```

Comment:
please also add a test for parameterised `InitVar`, eg. `InitVar[int]`

Reply:
I will test this. The type will be ignored by pydantic though. Init vars are passed to post_init and not part of the model. I could add a type check in pydantic_post_init, but that could turn out to be more complicated than the original fix.

Reply:
no need to check the type, we're really just testing that `field.type is not dataclasses.InitVar` still works (I think it won't, but could be wrong).

Reply:
field.type is not dataclasses.InitVar still worked, but now I have chosen to use dataclasses.fields instead. This also skips Class variables annotated with typing.ClassVar and it is a public function instead of the private `__dataclasses_fields__`.